### PR TITLE
feat: Analyse the data based on number of null values and repeated va…

### PR DESCRIPTION
**Pull request by StiVit**
Check some basic information about the df and output the info on a file for a better reading

### CODE REVIEW

File: src/word_analysis.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from wordcloud import WordCloud
+import numpy as np
+from textblob import TextBlob
+import matplotlib.pyplot as plt
+
+
+def generate_word_cloud(df, column):
```
Comment (grumpycatyo-collab) [inline src/word_analysis.py]: I don't think you need these methods as scripts in here
They are stuff for your research, that show what are the characteristics of your dataset and stuff

File: src/word_analysis.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    df[column] = df[column].fillna('')
+    text = "".join(cat for cat in df[column][:100])
+    word_cloud = WordCloud(collocations=False, background_color='white').generate(text)
+    plt.imshow(word_cloud, interpolation='bilinear')
+    plt.axis("off")
+    plt.show()
+
+
+def sentimental_analysis(text):
```
Comment (grumpycatyo-collab) [inline src/word_analysis.py:6]: Same here

File: utils/structured_output.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import logging
+import os
+import matplotlib.pyplot as plt
+from utils.logger_setup import setup_logger
+
+def structure_output(df):
```
Comment (grumpycatyo-collab) [inline utils/structured_output.py]: Same with this function
Mainly it doesnt give you any important stuff that you will use afterwards. Thats why u can use Jupyter Notebook for researching.

**Review by grumpycatyo-collab — CHANGES_REQUESTED** [review]
1. Can you make a folder called research where you will work with jupyter notebook and there you will show how your functions work. While they look pretty good, I don't see the outcome which is bad. 
2. Add the `.idea` folder to `.gitignore` please since it's only Pycharm stuff (same for `.DS_Store)
3. After doing your research, you proceed to make your scripts (the smallest part of your project btw) that will do only functional stuff (giving results, not only graphs and stuff)


For refference take a look at some of my stuff:
https://github.com/grumpycatyo-collab/animated-octo-couscous - here you can steal stuff from
https://github.com/grumpycatyo-collab/ubiquitous-test - work on a dataset
https://github.com/grumpycatyo-collab/heavy-rick-metal - my not finished sentiment analysis on rich and morty phrases
all this stuff is for research

File: .idea/.gitignore
Comment (grumpycatyo-collab) [inline .idea/.gitignore:1]: Delete these files pls

File: main.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from src.proccess_csv import process_data
+from sklearn.metrics.pairwise import cosine_similarity
+
+if __name__ == "__main__":
+    df_movies, columndictionary = process_data()
+    genre_columns = df_movies.columns[24:]
+
+    user_preference = [1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0]
```
Comment (grumpycatyo-collab) [inline main.py]: What is the user preference in this case?

File: src/word_analysis.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+from textblob import TextBlob
+import matplotlib.pyplot as plt
+
+def sentimental_analysis(text):
+    """
+    Creates an object of the TextBlob class, the sentiment attribute is used:
+    polarity: how positive or negative the sentiment is
+    """
+    sentiment = TextBlob(str(text)).sentiment
+    return sentiment.polarity, sentiment.subjectivity
```
Comment (grumpycatyo-collab) [inline src/word_analysis.py:12]: why do you need these ?

**Review by grumpycatyo-collab — COMMENTED** [review]
Just questions for now

File: main.py (hunk 4 above)
Comment (StiVit) [reply main.py]: The preferred genres, this is only a test for now, I'll add later to prompt the user the preferred genres and mask them to the array

File: src/word_analysis.py (hunk 5 above)
Comment (StiVit) [reply src/word_analysis.py:12]: I plan to add more features into the algorithm to make the outcome more accurate, this is the same reason why I made the algorithm for extracting features

**Review by grumpycatyo-collab — APPROVED** [review]
